packages/react/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,192 @@
+ * // returns [1, 'firstSpacer', 4, 5, 6, 'lastSpacer', 100]
+ *
+ * @example
+ * currentPage = 97, totalPages = 100, maxVisiblePages = 7
+ * // returns [1, 'firstSpacer', 96, 97, 98, 99, 100]
+ */
+
+function getRange(currentPage: number, totalPages: number, maxVisiblePages: number): Array<string | number> {
+  // the total amount of visible pages is whatever's lower, totalPages or maxVisiblePages
+  // maxVisiblePages has a lower limit of 5
+  const visiblePages = Math.min(totalPages, Math.max(maxVisiblePages, 5))
+
+  const min = 1
+  // the center part of the range starts with the current page minus half of the visible pages
+  let centerStartPage = currentPage - Math.floor(visiblePages / 2)
+  // centerStartPage has a lower limit of 1
+  centerStartPage = Math.max(centerStartPage, min)
+  // centerStartPage has an upper limit of 1 plus total pages minus visible pages
+  centerStartPage = Math.min(centerStartPage, min + totalPages - visiblePages)
+
+  const pages = Array.from({ length: visiblePages }, (_el, i) => centerStartPage + i).reduce<Array<string | number>>(
+    (acc, pageNr, index) => {
+      if (index === 0 && pageNr !== 1) {
+        return [1, 'firstSpacer']
+      }
+
+      if (totalPages > visiblePages && index === visiblePages - 2 && currentPage < totalPages - 2) {
+        return [...acc, 'lastSpacer', totalPages]
+      }
+      // Skip a number when spacer is already added
+      if ((acc.includes('firstSpacer') && index === 1) || (acc.includes('lastSpacer') && index === visiblePages - 1)) {
+        return acc
+      }
+      return [...acc, pageNr]
+    },
+    [],
+  )
+
+  return pages
+}
+
+export const Pagination = forwardRef(
+  (
+    {
+      className,
+      collectionSize,
+      onPageChange,
+      page = 1,
+      pageSize = 10,
+      maxVisiblePages = 7,
+      ...restProps
+    }: PaginationProps,
+    ref: ForwardedRef<HTMLElement>,
+  ) => {
+    const [currentPage, setCurrentPage] = useState(page)
+
+    // If page prop changes, set currentPage so component can be managed externally
+    useEffect(() => {
+      setCurrentPage(page)
+    }, [page])
+
+    const totalPages = Math.ceil(collectionSize / pageSize)
+
+    // Get array of page numbers and / or spacers
+    const range = getRange(currentPage, totalPages, maxVisiblePages)
+
+    const onChangePage = (newPage: number) => {
+      if (onPageChange !== undefined) {
+        onPageChange(newPage)
+      }
+      setCurrentPage(newPage)
+    }
+
+    const onPrevious = () => {
+      onChangePage(currentPage - 1)
+    }
+
+    const onNext = () => {
+      onChangePage(currentPage + 1)
+    }
+
+    // Don't show pagination if you only have one page
+    if (collectionSize <= pageSize) {
+      return null
+    }
+
+    return (
+      <nav {...restProps} className={clsx('amsterdam-pagination', className)} ref={ref} aria-label="Paginering">
+        <ol className="amsterdam-pagination__list">
+          <li>
+            <button
```

Comment:
Responding to the comment by @VincentSmedinga about links vs buttons here:

> These should totally be links from the start.

Hm, I think it's a bit more complicated than that. I did a quick inventory how other design systems and MDN does this, 5x buttons vs 7x links. I think it very much depends on the structure of your app. 

Most Amsterdam apps I checked use url params to indicate pagination, which makes it more of a filter I think? In those cases, buttons make more sense imo. These are amsterdam.nl itself (which does use links), data.amsterdam.nl and onderzoek.amsterdam.nl. Mijn.amsterdam.nl creates a url with `react-router-dom` I think, so something like /page/2 (which should be a link, maybe? But they use buttons).

Erfpacht has multiple Pagination components on one page, so links wouldn't make a lot of sense there. 

I think for use within Amsterdam a button makes more sense. Maybe we should support both, if it's needed.

Reply:
Just discussed this, for now we'll use buttons, because changing it to links makes the component a bit more complex. I imagine we need a variant with links as well, but we should test it thoroughly in an application setting before we add that.